src/wp-includes/class-wp-scripts.php

```diff
@@ -1082,10 +1148,19 @@ private function get_eligible_loading_strategy( $handle ) {
 		/*
 		 * Handle known blocking strategy scenarios.
 		 * - An empty strategy is synonymous with blocking.
-		 * - A script bundle (where $src is false) must always be blocking since the after inline script cannot be
+		 * - A script alias (where $src is false) must always be blocking since the after inline script cannot be
 		 *   delayed as there is no external script tag and thus no load event at which the inline script can be run.
 		 */
-		if ( empty( $intended_strategy ) || empty( $this->registered[ $handle ]->src ) ) {
+		if ( empty( $intended_strategy ) ) {
+			return '';
+		}
+
+		// Aliases that have before/after inline scripts can never be delayed since there is no load event.
+		if ( ! $this->registered[ $handle ]->src && (
+				$this->get_data( $handle, 'before' ) ||
+				$this->get_data( $handle, 'after' )
+			)
```

Comment:
This logic is duplicated with a condition above on line 1095. It should be extracted into a helper private method.

Reply:
Well, not exactly duplicated. Maybe some refactoring is in order.

Reply:
Refactored in e72fcc0a72